mlflow/pipelines/steps/data_clean.py

```diff
@@ -0,0 +1,93 @@
+import logging
+import os
+import time
+import importlib
+import sys
+from typing import Dict, Any
+
+from mlflow.pipelines.cards import BaseCard
+from mlflow.pipelines.step import BaseStep
+from mlflow.pipelines.utils.execution import get_step_output_path
+from mlflow.pipelines.utils.step import get_pandas_data_profile
+
+
+_logger = logging.getLogger(__name__)
+
+
+_INPUT_FILE_NAME = "dataset.parquet"
+_CLEANED_OUTPUT_FILE_NAME = "dataset_clean.parquet"
+
+
+class DataCleanStep(BaseStep):
+    def __init__(self, step_config: Dict[str, Any], pipeline_root: str):
+        super().__init__(step_config, pipeline_root)
+
+        self.run_end_time = None
+        self.execution_duration = None
+
+    def _build_profiles_and_card(self, clean_df) -> BaseCard:
+        # Build profiles for input dataset, and train / validation / test splits
+        clean_data_profile = get_pandas_data_profile(
+            clean_df.reset_index(drop=True),
+            "Profile of Cleaned Dataset",
+        )
+
+        # Build card
+        card = BaseCard(self.pipeline_name, self.name)
+        # Tab #1: data profile for cleaned data:
+        card.add_tab("Cleaned Data Profile", "{{PROFILE}}").add_pandas_profile(
+            "PROFILE", clean_data_profile
+        )
+        # Tab #2: run summary.
+        (
+            card.add_tab(
+                "Run Summary",
+                """
+                {{ SCHEMA_LOCATION }}
+                {{ CLEAN_DATA_NUM_ROWS }}
+                {{ EXE_DURATION}}
+                {{ LAST_UPDATE_TIME }}
+                """,
+            ).add_markdown(
+                "CLEAN_DATA_NUM_ROWS", f"**Number of cleaned dataset rows:** `{len(clean_df)}`"
+            )
+        )
```

Comment:
Maybe remove this tab for now, and add this tab back as the values become available/written?

Reply:
So the preprocess step i think is almost ready.  I need to add tests but these values should be real at this point.